appcues/src/main/java/com/appcues/analytics/ExperienceLifecycleTracker.kt

```diff
@@ -38,12 +40,20 @@ internal class ExperienceLifecycleTracker(
     private val storage: Storage by inject()
     private val appcuesCoroutineScope: AppcuesCoroutineScope by inject()
 
-    fun start(stateMachine: StateMachine, dispatcher: CoroutineDispatcher = Dispatchers.IO) {
-        appcuesCoroutineScope.launch(dispatcher) { stateMachine.observeState() }
-        appcuesCoroutineScope.launch(dispatcher) { stateMachine.observeErrors() }
+    var stateJob: Job? = null
+    var errorJob: Job? = null
+
+    private fun cancelJobs() {
```

Comment:
cancel both jobs if we hit the end of the experience or an error.  (maybe we should only cancel on EndExperience) ?

Reply:
tricky one. I think there is the potential case where a fatal error occurs and it does _not_ go through EndExperience state - just right to Idle, so we'd want to catch that one, like you have.

There is also the other case where an error can occur but then be recovered - does this pose an issue now?

Reply:
upon testing I found out this is probably not right, an ExperienceEnding should not cancel those jobs too, as new experiences can still be processed.